specs/liberty/api.rst

```diff
@@ -0,0 +1,203 @@
+  https://blueprints.launchpad.net/gslb/+spec/awesome-thing should be named
+  awesome-thing.rst .  Please do not delete any of the sections in this
+  template.  If you have nothing to say for a whole section, just write: None
+  For help with syntax, see http://sphinx-doc.org/rest.html
+  To test out your formatting, see http://www.tele3.cz/jbar/rest/rest.html
+
+=====
+ API 
+=====
+
+https://blueprints.launchpad.net/gslb/+spec/api
+
+We need an API, so defining it is a good start.
+
+Example JSON Snippets
+=====================
+
+Load Balancer JSON Snippet
+--------------------------
+
+ .. code-block:: json
+
+    {
+      "id": "f9fc6210-8efa-4492-9b4f-d87ba98f35b8",
+      "name": "Main Website",
+      "fqdn": "www.gslb.example.com.",
+      "domain_name": "example.com.",
+      "flavor": "example_1",
+      "status": "ACTIVE",
+      "links": {
+        "self": "http://example.gslb.openstack.org/v0.1/gslbs/f9fc6210-8efa-4492-9b4f-d87ba98f35b8",
+        "pools": "http://example.gslb.openstack.org/v0.1/gslbs/f9fc6210-8efa-4492-9b4f-d87ba98f35b8/pools",
+        "status": "http://example.gslb.openstack.org/v0.1/gslbs/f9fc6210-8efa-4492-9b4f-d87ba98f35b8/status"
+      }
+    }
+
+
+Pool JSON Snippet
+-----------------
+
+ .. code-block:: json
+
+    {
+      "id": "f9fc6210-8efa-4492-9b4f-d87ba98f35b8",
+      "name": "Main Website Pool",
+      "flavor": "example_1",
+      "status": "ACTIVE",
+      "members":["SEE MEMBER SNIPPETS BELOW"],
+      "links": {
+        "self": "http://example.gslb.openstack.org/v0.1/pools/f9fc6210-8efa-4492-9b4f-d87ba98f35b8",
+        "pool_members": "http://example.gslb.openstack.org/v0.1/pools/f9fc6210-8efa-4492-9b4f-d87ba98f35b8/pool_members",
+        "status": "http://example.gslb.openstack.org/v0.1/pools/f9fc6210-8efa-4492-9b4f-d87ba98f35b8/status"
+      }
+    }
+
+Pool Member JSON Snippet - Neutron LBaaS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id": "f9fc6210-8efa-4492-9b4f-d87ba98f35b8",
+      "name": "Main Website",
+      "type": "neutron_lbaas_v2",
+      "region": "RegionTwo",
+      "neutron_id": "df40954c-982a-4778-a1b2-32b93cf75af9",
+      "status": "ACTIVE",
+      "endpoints": [
+        "10.10.0.1"
+      ]
+    }
+
+
+Pool Member JSON Snippet - Neutron Port
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id": "f9fc6210-8efa-4492-9b4f-d87ba98f35b8",
+      "name": "Main Website",
+      "type": "neutron_port",
+      "region": "RegionTwo",
+      "neutron_id": "78fa9c7f-f200-49df-912b-ae4679fd21e9",
+      "status": "ACTIVE",
+      "endpoints": [
+        "10.10.0.1"
+      ]
+    }
+
+Pool Member JSON Snippet - IP
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id": "f9fc6210-8efa-4492-9b4f-d87ba98f35b8",
+      "name": "Main Website",
+      "type": "IP",
+      "status": "ACTIVE",
+      "endpoints": [
+        "10.10.0.1"
+      ]
+    }
+
```

Comment:
We need JSON snippets for the monitors and/or health checks.

We also need to decide if the members should have port/protocol information of if that is only visible in the monitor/health check JSON